nix/default.nix

```diff
@@ -27,27 +24,33 @@ let
     ++ iohkNix.overlays.iohkNix
     # our own overlays:
     ++ [
-      (pkgs: _: with pkgs; {
-
-        # commonLib: mix pkgs.lib with iohk-nix utils and our own:
-        commonLib = lib // iohkNix
-        // import ./util.nix { inherit lib haskell-nix; }
-        # also expose our sources and overlays
-        // { inherit overlays sources; };
-      })
+      # Modifications to derivations from nixpkgs
       (import ./overlays/nixpkgs-overrides.nix)
       # This contains musl-specific stuff, but it's all guarded by appropriate host-platform
       # checks, so we can include it unconditionally
       (import ./overlays/musl.nix)
       # add pre-commit-hooks which isn't available in 20.03
       (import ./overlays/pre-commit-hooks.nix)
+      # add easyPS
+      (import ./overlays/easy-purescript.nix)
+      # sphinx haddock support
+      (import ./overlays/sphinx-haddock.nix)
+      # nodejs headers  (needed for purescript builds)
+      (import ./overlays/nodejs.nix)
     ];
 
-  pkgs = import nixpkgs {
+  pkgs = import sources.nixpkgs {
     inherit system crossSystem;
     overlays = extraOverlays ++ overlays;
     config = haskellNix.config // config;
   };
 
+  pkgsMusl = import sources.nixpkgs {
+    system = "x86_64-linux";
+    crossSystem = pkgs.lib.systems.examples.musl64;
+    overlays = extraOverlays ++ overlays;
+    config = haskellNix.config // config;
+  };
+
 in
-pkgs
+pkgs // { inherit pkgsMusl; }
```

Comment:
I actually don't love the fact that `nix/default.nix` "is" a nixpkgs (this code was written by someone else ;) ). If anything, it should be "all the stuff defined in `nix` (we don't have to stuff *everything* in overlays either). 

So why don't we just return an attribute set here: at least containing `pkgs`, and `pkgsMusl`, and maybe some of the other stuff.

Reply:
Or why not:
```
pkgsLocal // { inherit pkgs pkgsMusl; }
```
i.e. it's got all the local stuff and also pkgs and pkgsMusl?

Reply:
Yep, that was a good suggestion and i followed it (albeit with slightly changed attribute set)